src/main/java/htsjdk/samtools/reference/FastaSequenceIndexCreator.java

```diff
@@ -0,0 +1,179 @@
+            FaiEntryBuilder entry = null;
+
+            // read the lines two by two
+            for (String line = in.readLine(); previous != null; line = in.readLine()) {
+                // in this case, the previous line contains a header and the current line the first sequence
+                if (previous.charAt(0) == '>') {
+                    // first entry should be skipped
+                    if (entry != null) {
+                        // add the entry to the index
+                        index.add(entry.build());
+                    }
+                    // update the sequence index
+                    sequenceIndex++;
+                    entry = new FaiEntryBuilder(sequenceIndex, previous, line, location);
+                } else if (line != null && line.charAt(0) == '>') {
+                    // update the location, next iteration the sequence will be handled
+                    location = in.getPosition();
+                } else if (line != null && !line.isEmpty()) {
+                    // update in case it is not a blank-line
+                    entry.updateWithSequence(line);
+                }
+                // set the previous to the current line
+                previous = line;
+            }
+
+            // add the last entry
+            index.add(entry.build());
+
+            // and return the index
+            return index;
+        }
+    }
+
+    // utility class for building the FastaSequenceIndexEntry
+    private static class FaiEntryBuilder {
+        private final int index;
+        private final String contig;
+        private final long location;
+        // the bytes per line is the bases per line plus the end of the line
+        private final int basesPerLine;
+
+        // the size is updated for each line in the input using updateWithSequence
+        private long size;
+        // flag to check if the supposedly last line was already reached
+        private boolean lessBasesFound;
+
+        private FaiEntryBuilder(final int index, final String header, final String firstSequenceLine, final long location) {
+            if (header == null || header.charAt(0) != '>') {
+                throw new SAMException("Wrong sequence header: " + header);
+            } else if (firstSequenceLine == null) {
+                throw new SAMException("Empty sequences could not be indexed");
+            }
+            this.index = index;
+            // parse the contig name (without the starting '>')
+            this.contig =  header.substring(1);
+            this.location = location;
+            this.basesPerLine = firstSequenceLine.length();
+
+            this.size = firstSequenceLine.length();
+            this.lessBasesFound = false;
+        }
+
+        private void updateWithSequence(final String sequence) {
+            if (sequence.length() > basesPerLine) {
+                throw new SAMException(String.format("Sequence bases for '{}' exceed the maximum length ({}): {}",
```

Comment:
The message here is technically correct, but I think I would find it confusing, since other than the last line all lines should be the same length, not variable lengths with a maximum.  How about "Sequence line for {} was longer than the expected length ({}): {}"